File: src/main/java/graph/edge/SimplexChannelFactory.java
package graph.edge;

import graph.vertex.Vertex;
import org.jetbrains.annotations.NotNull;

public class SimplexChannelFactory implements ChannelFactory {
    @Override
    public Edge createChannel(@NotNull Vertex from, @NotNull Vertex to, int weight, double errors) {
        return new SimplexChannel(from, to, weight, errors);
    }

    @Override
    public String toString() {
        return "Simplex";
    }
}

File: src/main/java/mvc/model/channel/SimplexChannelFactory.kt
package mvc.model.channel

import mvc.model.node.Node

class SimplexChannelFactory : ChannelFactory {
    override fun createChannel(from: Node, to: Node, weight: Int, errors: Double): Channel {
        return SimplexChannel(from, to, weight, errors)
    }

    override fun toString(): String {
        return "Симплексний канал"
    }
}
